### PR TITLE
Initial outline for ens beginners guide

### LINKED ISSUE #23: Create a beginners guide to ENS
_This could be part one of a series, but let's start with this first part._

**Word count**: 500-800 words
**Deadline**: 1st April

Create an end-to-end blog post covering how to incorporate ENS into a reasonably real-world Dapp. This should focus on practical code examples and an application that people can take and build while following the tutorial.

The post should use clear technical language, reducing passive voice, future tense, weasel words etc, I am happy to offer advice on what all this means. 🙂

Focus on what it is someone gets out of reading the post, why they would want to read it, and what they are trying to accomplish by reading it. Again happy to answer questions.

### PR DESCRIPTION
**Pull request by darrenlangley**
@ChrisChinchilla 

I have pulled together an outline of the ENS beginner's guide. Let me know what you think?

I am scoping out the code example. There are a couple of complications that I need to work through.

I wanted to implement a tipper service or transferring funds to an ENS name. The issue is that we have to use a public registry (mainnet/testnet, over Infura would be simplest) to be able to look up names so we can't be sure the account they use will have funds. So we may only be able demo the lookup instead of doing the transfer.

Having a UI makes the example realistic but obviously bulky. I could implement a simple React app. I wanted to implement an address text box that dynamically looks up the ENS name. Just want to make sure that isn't overkill? In theory, not a problem but slightly risky in terms of time. 

An easier solution would be to implement a simple CLI that returns the ENS details of the address you provide as an argument, something like:
```
> enslookup ethereum.eth
```
Which returns the address, content, and pubkey of the ENS name. The benefit is that it shows that multple targets can be referenced.

Let me know your thoughts?
 
resolves #23

### CODE REVIEW

File: Outlines/ens-beginners-guide.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+- Each name can point to a single target or multiple targets depending on what you want to achieve. 
+- It is highly extensible and supports many target types such as Ethereum addresses/contracts, ABI definitions, content hashes (IPFS), public keys, and key/value text items
+- It lives on the Ethereum blockchain so is naturally decentralised.
+- It can be queried from smart contracts
+- By being able to resolve friendly names it enhances your application's usability.
+- Particularly good for reliably resolving contract addresses and their associated ABIs
+
+## Code Walkthrough
+
+- Many Ethereum libraries, such as Web3 or Ethers.js now support ENS out-of-the-box.
```
Comment (ChrisChinchilla) [inline Outlines/ens-beginners-guide.md:17]: @darrenlangley For now let's stick with Web3.js.

File: Outlines/ens-beginners-guide.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# ENS: Beginner's Guide
```
Comment (ChrisChinchilla) [inline Outlines/ens-beginners-guide.md:1]: @darrenlangley Let's make sure to start with the problem, which is something like… Highly paraphrased of course:

Ethereum addresses are long, unwieldy, and hard to remember. ENS is like DNS, in that it maps a memorable shortcut to that address.

That is what you have said in the points below, but let's make the problem we are showing people how to solve, the first and foremost point.

**Comment by ChrisChinchilla** [comment]
> I wanted to implement a tipper service or transferring funds to an ENS name. The issue is that we have to use a public registry (mainnet/testnet, over Infura would be simplest) to be able to look up names so we can't be sure the account they use will have funds. So we may only be able demo the lookup instead of doing the transfer.

How can I help you solve this? We can create an Infura endpoint with ETH, is Rinkeby ETH enough?

**Comment by ChrisChinchilla** [comment]
> Having a UI makes the example realistic but obviously bulky. I could implement a simple React app. I wanted to implement an address text box that dynamically looks up the ENS name. Just want to make sure that isn't overkill? In theory, not a problem but slightly risky in terms of time.

I don't mind, you can also make this a CLI application, or go for a MVP approach, we always add more in future tutorials.

File: Outlines/ens-beginners-guide.md (hunk 1 above)
Comment (darrenlangley) [reply Outlines/ens-beginners-guide.md:17]: All good.

File: Outlines/ens-beginners-guide.md (hunk 2 above)
Comment (darrenlangley) [reply Outlines/ens-beginners-guide.md:1]: Thanks will do.

**Review by ChrisChinchilla — APPROVED** [review]

**Pull request merged**